src/server/document-controller.js

```diff
@@ -72,7 +74,7 @@ class DocumentController {
         throw new NotFound();
       }
 
-      if (slug !== doc.slug) {
+      if (slug !== UNDEFINED_SLUG && slug !== doc.slug) {
```

Comment:
Can be reverted if you use my suggestion above

Reply:
Reverted